### PR TITLE
Pkgconfig support

### PR DESCRIPTION
**Pull request by tintou**

### CODE REVIEW

File: src/lib/editorconfig.pc.in [hunk 1]
```diff
@@ -0,0 +1,9 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+libdir=${prefix}/@CMAKE_INSTALL_LIBDIR@
+includedir=${prefix}/@CMAKE_INSTALL_INCLUDEDIR@/editorconfig
+ 
+Name: editorconfig
+Description: Configuration file format for defining coding styles in shared projects
```
Comment (xuhdev) [inline src/lib/editorconfig.pc.in]: This line may be confusing -- the package itself is not a configuration file. Maybe

    EditorConfig C Core: A library for handling EditorConfig files, a configuration file format for defining coding styles in shared projects.

Do you think `editorconfig-core-c` is a better name?

File: CMakeLists.txt [hunk 2]
```diff
@@ -24,17 +24,13 @@
 # POSSIBILITY OF SUCH DAMAGE.
 #
 
-cmake_minimum_required(VERSION 2.8.7)
-cmake_policy(VERSION 2.8.7)
+cmake_minimum_required(VERSION 3.0)
+cmake_policy(VERSION 3.0)
```
Comment (xuhdev) [inline CMakeLists.txt]: Is this change necessary? The reason I hesitate on this is because Ubuntu 14.04 still has an older version of cmake (2.8.12). I would prefer to upgrade after it retires.

**Comment by tintou** [comment]
@xuhdev Ah yeah, it's completely possible to not depend on CMake 3.0, I just wanted to use concise variables available since the introduction of the `project` command. Here is an updated version that doesn't change the CMake requirement.
I have taken your description and just shrunk it a little bit
Regarding the library name, as pkgconfig mostly targets C/C++ it's probably worth being concise here :man_shrugging: also I never found mention of the `core` name (the .so name and headers don't mention it) so it's probably better to keep this like that for discoverability

**Comment by xuhdev** [comment]
Looks good to me. Thanks!
